pkg/collector/corechecks/servicediscovery/usm/ruby_test.go

```diff
@@ -0,0 +1,48 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+package usm
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGenerateNameFromRailsApplicationRb(t *testing.T) {
+	tests := []struct {
+		name     string
+		path     string
+		expected string
+	}{
+		{
+			name:     "name is found",
+			path:     "./testdata/application.rb",
+			expected: "rails_hello",
+		},
+		{
+			name:     "name not found",
+			path:     "./testdata/application_invalid.rb",
+			expected: "",
+		},
```

Comment:
as my other comment suggested, let's work on the test cases to gain full coverage of the function